sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveDirCommand.scala

```diff
@@ -108,8 +109,11 @@ case class InsertIntoHiveDirCommand(
         outputLocation = tmpPath.toString)
 
       if (overwrite && fs.exists(writeToPath)) {
+        val isTrashEnabled = sparkSession.sessionState.conf.trashEnabled
         fs.listStatus(writeToPath).foreach { existFile =>
-          if (Option(existFile.getPath) != createdTempDir) fs.delete(existFile.getPath, true)
+          if (Option(existFile.getPath) != createdTempDir) {
+            Utils.moveToTrashOrDelete(fs, existFile.getPath, isTrashEnabled, hadoopConf)
+          }
```

Comment:
I am doing this too.   this pr (such as INSERT OVERWRITE DIRECTION part ) can avoid our user write wrong  dir path that have data (such as DB's path, happened before).
Move data to trash can make recovery of production data faster in the event of such a disaster.
FYI @maropu